### PR TITLE
Make REST Client a bean when the scope property is set in config

### LINKED ISSUE #37007: REST client reactive should allow to instantiate clients based on (external) classes without `@RegisterRestClient` annotation
### Description

During the migration from rest-client to rest-client-reactive, it has turned out, that the reactive doesn't allow to instantiate clients based on external (imported from dependencies) interface definitions (lacking the `@RegisterRestClient` annotation).

In the rest-client classic, this configuration alone is sufficient to instantiate a client bean (providing that `MyRestServiceClass` contains appropriate JAX-RS annotations):

```yaml
rest-client:
  "myPackage.MyRestServiceClass":
    scope: jakarta.enterprise.context.Singleton
    url: ${my-service.url}
```

### Implementation ideas

It was discussed in the Zulip chat, and suggested by @Ladicek, that such a functionality (automatically adding `@RegisterRestClient` during build time to classes injected into `@RestClient`-annotated slots), could be implemented using a build step.

### PR DESCRIPTION
**Pull request by geoand**
- Closes: #37007

### CODE REVIEW

File: extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientBuildConfig.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+@ConfigGroup
+public class RestClientBuildConfig {
+
+    /**
+     * The CDI scope to use for injection. This property can contain either a fully qualified class name of a CDI scope
+     * annotation (such as "jakarta.enterprise.context.ApplicationScoped") or its simple name (such as
+     * "ApplicationScoped").
```
Comment (cescoffier) [inline extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientBuildConfig.java:14]: I would indicate what's the default if not set.

**Review by cescoffier — APPROVED** [review]
just a comment on the javadoc of the new property. I think we should indicate the default if unset.

File: extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientBuildConfig.java (hunk 1 above)
Comment (geoand) [reply extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientBuildConfig.java:15]: Done

**Comment by geoand** [comment]
Happens to me everytiime :)

**Pull request merged**